From 6a2490239e1f86c8009ab192e2361aa4f911da2e Mon Sep 17 00:00:00 2001
From: He Sheng <hesheng@wxiat.com>
Date: Fri, 27 May 2022 10:01:54 +0800
Subject: [PATCH] sw64: fix ptrace.h with types.h and NOT __ASSEMBLY__
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: 6a2490239e1f86c8009ab192e2361aa4f911da2e
Modified-by-SEL: No


Sunway inclusion
category: feature
bugzilla: https://gitee.com/openeuler/kernel/issues/I56OLG

--------------------------------

It may report compile errors on some user apps if linux/types.h is
nowhere to be included. And a __ASSEMBLY__ guard makes it work if
included by assembly codes.

Signed-off-by: He Sheng <hesheng@wxiat.com>

Signed-off-by: Gu Zitao <guzitao@wxiat.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 arch/sw_64/include/uapi/asm/ptrace.h | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/arch/sw_64/include/uapi/asm/ptrace.h b/arch/sw_64/include/uapi/asm/ptrace.h
index a45b10b8914a..80bad067fc15 100644
--- a/arch/sw_64/include/uapi/asm/ptrace.h
+++ b/arch/sw_64/include/uapi/asm/ptrace.h
@@ -2,6 +2,9 @@
 #ifndef _UAPI_ASM_SW64_PTRACE_H
 #define _UAPI_ASM_SW64_PTRACE_H
 
+#include <linux/types.h>
+
+#ifndef __ASSEMBLY__
 /*
  * User structures for general purpose, floating point and debug registers.
  */
@@ -21,6 +24,7 @@ struct user_fpsimd_state {
 	__u64 fpcr;
 	__u64 __reserved[3];
 };
+#endif
 
 /* PTRACE_ATTACH is 16 */
 /* PTRACE_DETACH is 17 */
-- 
2.33.0

